frontend/src/metabase/query_builder/components/expressions/Editor/Editor.tsx

```diff
@@ -222,8 +234,18 @@ function useExpression<S extends StartRule = "expression">({
         onChange(clause);
       }
       onError(null);
-      setHasChanges(false);
-      setSource(formatExpression(expression));
+      setIsFormatting(true);
+      formatExpression(expression)
```

Comment:
Due to `prettier` being async `formatExpression` becomes async. This is unfortunate, but it usually resolves in faster that React can render.